src/NuGet.Core/NuGet.Configuration/Proxy/ProxyCache.cs

```diff
@@ -99,6 +99,14 @@ public WebProxy GetUserConfiguredProxy()
                 {
                     webProxy.Credentials = new NetworkCredential(userName, password);
                 }
+                
+                var noProxy = _settings.GetValue(SettingsUtility.ConfigSection, ConfigurationContants.NoProxy);
```

Comment:
lets commonize this into one method? And also does this need a documentation change?